src/eval_context.rs

```diff
@@ -1521,8 +1527,9 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 //let src = adt::MaybeSizedValue::sized(src);
                 //let dst = adt::MaybeSizedValue::sized(dst);
                 let src_ptr = match src {
-                    Value::ByRef(ptr) => ptr,
-                    _ => bug!("expected pointer, got {:?}", src),
+                    Value::ByRef(ptr, true) => ptr,
+                    // TODO: Is it possible for unaligned pointers to occur here?
```

Comment:
Oh... apparently it's more involved http://play.integer32.com/?gist=8b0f3ea2bf1500765fafaf0aadfb81bc&version=nightly